### PR TITLE
Compose GalleryListScene

### PR DESCRIPTION
**Pull request by VDostoyevskiy**

### CODE REVIEW

**Comment by VDostoyevskiy** [comment]
ProvideSideSheetContent need more robust design
To handle two screens both on composition

**Comment by FooIbar** [comment]
> ProvideSideSheetContent need more robust design To handle two screens both on composition

Cancel the CI after merging this if you're going to do that in another PR, as it's not ready to be pushed to users.

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryListFragment.kt [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+            }
+        }
+
+        val listMode by remember {
+            Settings.listModeBackField.valueFlow()
+        }.collectAsState(Settings.listMode)
+        val marginH = dimensionResource(id = R.dimen.gallery_list_margin_h)
+        Box(
+            modifier = Modifier.fillMaxSize().scale(animatedSearchLayout).alpha(animatedSearchLayout),
+            contentAlignment = Alignment.Center,
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryListFragment.kt:509]: The alignment of the list should be TopCenter.

**Comment by FooIbar** [comment]
TODO:
Side sheet item drag and drop
Side sheet item swipe to dismiss
Append and prepend loading indicator & retry button

File: app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt [hunk 2]
```diff
@@ -362,10 +363,10 @@ class MainActivity : EhActivity() {
                     SideDrawer(
                         drawerContent = {
                             if (sheet != null) {
+                                val insets = WindowInsets.systemBars.only(WindowInsetsSides.Top + WindowInsetsSides.End)
                                 ModalDrawerSheet(
-                                    modifier = Modifier.widthIn(max = (configuration.screenWidthDp - 112).dp),
+                                    modifier = Modifier.widthIn(max = (configuration.screenWidthDp - 112).dp).windowInsetsPadding(insets),
                                     drawerShape = ShapeDefaults.Large.copy(topEnd = CornerSize(0), bottomEnd = CornerSize(0)),
-                                    windowInsets = WindowInsets.systemBars.only(WindowInsetsSides.Top + WindowInsetsSides.End),
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt:375]: This ain't the way.
`ModalDrawerSheet` takes vertical system bars insets by default and pad content for them.
If you want side sheet to be edge-to-edge, you should handle the insets in the lazy list;
if you don't, consume bottom insets as well in windowInsetsPadding.

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt:375]: Ah, I think we should limit SideSheet height as well
2/3 screen width?

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt:375]: > Ah, I think we should limit SideSheet height as well 2/3 screen height?

Why? I think side sheet height is 100% in M3 specs.

**Pull request merged**